hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -520,13 +520,14 @@ public void accept(Packet packet) {
         public void accept(Packet packet) {
             // currently service registration is done after the creation of the packet dispatcher,
             // hence we need to lazily initialize the JetPacketConsumer
-            if (packetConsumer == null) {
-                packetConsumer = serviceManager.getService(JET_SERVICE_NAME);
-                if (packetConsumer == null) {
+            Consumer<Packet> localPacketConsumer = this.packetConsumer;
```

Comment:
we could probably avoid volatiles here altogether by adding a `getJetPacketConsumer()` to NodeExtension perhaps?